### ISSUE TITLE
Better error messages

### ISSUE THREAD

**Issue by MrTomRod** [LM-UGent/GenDisCal#1]
Hi!

I really like your tool and how incredibly fast it is!

I noticed three small issues:

- For fun, I ran GenDisCal on a protein fasta, which led to this output:
```
Reading signatures...
Segmentation fault (core dumped)
```

- One of my assemblies had lower case letters, and the output was:
```
Reading signatures...
2/2 (100%) files loaded 
Finished reading signatures (0 hr 0 min 0 sec)
File1,File2,Expected_Relation,Distance
A Total of 1 comparisons were performed (0 hr 0 min 0 sec)
```

- When I using (`--filelist`) and formatting as `--distancematrix`, the lower-cased FASTAs always had `nan` as output, but the return code was still 0.

In all of these cases, I would have preferred an error message and exit status 1. (Better design by contract.) It would also be nice if the error message could specify what went wrong.

This would make automatizing it easier because errors would be catched earlier. I now have to write code that checks if the output is valid.

**Comment by GlebGoussarov** [comment]
Apologies for the incredibly late reply. I somehow missed your message. To adress the comments in order:
- GenDisCal is not designed for protein fasta. I do not currently plan to add this functionality, but I will add some checks in the next version.
- For point 2 & 3, this should have been corrected as of v1.2.1

For the "bad" error messages, I am currently working on a massive redesign which should prevent these kinds of issues.

**Comment by MrTomRod** [comment]
Cool, thanks! Looking forward to future versions!
